recipe/meta.yaml

```diff
@@ -20,14 +20,19 @@ build:
   number: {{ build }}
   skip: true  # [win]
   {% set mpi_prefix = "mpi_" + mpi %}
+  {% set mpi_netmod = "" %}
+  {% if mpi == "mvapich" %}
+  {% set mpi_netmod = "_" + netmod %}
+  {% endif %}
+  {% if (mpi == "mvapich" and target_platform not in ["linux-64", "linux-aarch64", "linux-ppc64le"]) %}
+  skip: true
+  {% endif %}
+  {% set mpi_prefix = "mpi_" + mpi + mpi_netmod %}
```

Comment:
@xylar Yes, until MPIch and OpenMPI have made a decision regarding how to deal with OFI/UCX it is unlikely to work

I started to look into this feedstock because we have issues with the current libpnetcdf builds with OpenMPI and MPIch

Are you happy with the performances you are getting?

On what sort of platform do you use it?

I do not fully understand for instance why you defined for OpenMPI environmental variables like `export OMPI_MCA_btl=self,tcp` inside the recipe

Reply:
Thanks for the background.

I never use MPI from conda for anything where I expect performance, I have to admit.  Just for convenience and for toy problems on laptops or login nodes of HPC systems.

I'm primarily a python developer at this point but where I use MPI codes, I always build them with system MPI on our HPC systems (sometimes using spack).

Reply:
As to the `OMPI` environment variables, you'd have to ask @zklaus.  He added them in https://github.com/conda-forge/libpnetcdf-feedstock/pull/3.

Reply:
It seems that there are issues with the `linux-aarch64` and `linux-ppc64le` emulation **during the make check or ptest** (i.e., not during he build itself)  and more or less random errors/warnings like:
```
A requested component was not found, or was unable to be opened.  This
means that this component is either not installed or is unable to be
used on your system (e.g., sometimes this means that shared libraries
that the component requires are unable to be found/loaded).  Note that
PMIx stopped checking at the first component that it did not find.

Host:      4a229425862b
Framework: plm
Component: isolated
```

also

```
[][mvp_get_arch_type] Failed to automatically detect the CPU architecture.
[][mvp_get_arch_type] This may lead to subpar communication performance.
```

or

```
WARNING: Open MPI failed to get or set flags on a TCP socket.  This
should not happen.

This may cause unpredictable behavior, and may end up hanging or
aborting your job.

  Local host: 4a229425862b
  PID:        15390
  Error:      Bad file descriptor (9)
--------------------------------------------------------------------------
2 more processes have sent help message help-mpi-btl-tcp.txt / socket flag fail
2 more processes have sent help message help-mpi-btl-tcp.txt / server getpeername failed
1 more process has sent help message help-mpi-btl-tcp.txt / socket flag fail
```
These tests sometimes fail if there are not enough processors available, or remain stuck forever:

```
##[error]The job running on agent Azure Pipelines 152 ran longer than the maximum time of 360 minutes. For more information, see https://go.microsoft.com/fwlink/?linkid=2077134
```

It is very likely that this depends on the network and/or type of hardware that is available, something over which I do not think we have any control, and therefore **I suggest to only perform the `make check/ptest` for the `linux-64` platform for now** (most builds we do not it at all anyway).